scipy/stats/stats.py

```diff
@@ -3129,10 +3130,29 @@ class F_onewayConstantInputWarning(RuntimeWarning):
     def __init__(self, msg=None):
         if msg is None:
             msg = ("Each of the input arrays is constant;"
-                    "the F-value is not defined or infinite")
+                   "the F statistic is not defined or infinite")
         self.args = (msg,)
 
 
+class F_onewayBadInputSizesWarning(RuntimeWarning):
+    """
+    Warning generated by `f_oneway` when an input has length 0,
+    or if all the inputs have length 1.
+    """
+    pass
+
+
+def _create_f_oneway_nan_result(shape):
+    shp = shape[1:]
+    if shp == ():
+        f = np.nan
+        prob = np.nan
+    else:
+        f = np.full(shp, fill_value=np.nan)
+        prob = f.copy()
+    return F_onewayResult(f, prob)
+
+
 def f_oneway(*args):
```

Comment:
do we want to add an `axis=0` argument? Two reasons I can think of: 
(i) explicit better than implicit 
(ii) allows someone using data stored in pandas dataframes (or other similar data structures) to do anova on the columns. 

Even though we don't have that as a default, it seems like (ii) would be a common enough use case that we'd want to support that via a non-default value of `axis`. 

Is there a reason why we don't support a non `axis=0` that I'm not noticing at the moment?

Reply:
The only reason for not supporting an `axis` parameter is probably that no one ever implemented it.  I'm working on adding it now.